packages/ckeditor5-markdown-gfm/src/pastefrommarkdownexperimental.ts

```diff
@@ -125,23 +121,12 @@ export default class PasteFromMarkdownExperimental extends Plugin {
 		return withoutBodyTag.replace( /^<!--StartFragment-->/, '' ).replace( /<!--EndFragment-->$/, '' ).trim();
 	}
 
-	/**
-	 * Removes a single HTML wrapper tag from string.
-	 *
-	 * @private
-	 * @param htmlString Clipboard content without any OS specific tags.
-	 */
-	private _removeWrapperTag( htmlString: string ): string {
-		return htmlString.replace( /^<(\w+)\b[^>]*>\s*([\s\S]*?)\s*<\/\1>/, '$2' ).trim();
-	}
-
 	/**
 	 * Removes multiple HTML wrapper tags from a list of sibling HTML tags.
 	 *
-	 * @private
 	 * @param htmlString Clipboard content without any OS specific tags.
 	 */
-	private _removeListWrapperTagsAndBrs( htmlString: string ): string {
+	private _removeFirstLevelWrapperTagsAndBrs( htmlString: string ): string {
 		const tempDiv = document.createElement( 'div' );
```

Comment:
Is this safe in the context of interactive content? Maybe we should use a separate DOM document like here: https://github.com/ckeditor/ckeditor5/blob/7a5a239de7a46080626d02a27572f5199c1839b9/packages/ckeditor5-engine/src/dataprocessor/basichtmlwriter.ts#L21-L28

Reply:
Or maybe we should use the `DOMParser`? https://github.com/ckeditor/ckeditor5/blob/7a5a239de7a46080626d02a27572f5199c1839b9/packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.ts#L122C3-L122C72

Reply:
using `DOMParser`.